### PR TITLE
Pipe: fix drop pipe stuck (make connector subtask close faster by early returning)

### PR DESCRIPTION
**Pull request by DanielWang2035**
If we drop a pipe when it is transferring a large TsFile, the "drop pipe" statement will stuck for a while, until the TsFile is fully transferred or the query times out. This PR will make connector subtask close faster by early returning.

### CODE REVIEW

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/task/subtask/connector/PipeConnectorSubtask.java [hunk 1]
```diff
@@ -118,20 +122,28 @@ protected synchronized boolean executeOnce() {
 
       releaseLastEvent(true);
     } catch (PipeConnectionException e) {
-      throw e;
+      if (!isClosed.get()) {
+        throw e;
+      }
```
Comment (SteveYurongSu) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/task/subtask/connector/PipeConnectorSubtask.java:127]: else {
  // log something & releaseLastEvent()
}

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/task/subtask/connector/PipeConnectorSubtask.java [hunk 2]
```diff
@@ -118,20 +122,28 @@ protected synchronized boolean executeOnce() {
...
-              + "whether the implementation of PipeConnector is correct "
-              + "according to the pipe-api description.",
-          e);
+      if (!isClosed.get()) {
+        throw new PipeException(
+            "Error occurred during executing PipeConnector#transfer, perhaps need to check "
+                + "whether the implementation of PipeConnector is correct "
+                + "according to the pipe-api description.",
+            e);
+      }
```
Comment (SteveYurongSu) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/task/subtask/connector/PipeConnectorSubtask.java:135]: else {
  // log something & releaseLastEvent()
}

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/task/subtask/connector/PipeConnectorSubtask.java [hunk 3]
```diff
@@ -118,20 +122,28 @@ protected synchronized boolean executeOnce() {
...
     }
 
     return true;
   }
 
   @Override
   public void onFailure(@NotNull Throwable throwable) {
+    if (isClosed.get()) {
+      return;
```
Comment (SteveYurongSu) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/task/subtask/connector/PipeConnectorSubtask.java:144]: // log something & releaseLastEvent()

Comment (SteveYurongSu) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/task/subtask/connector/PipeConnectorSubtask.java:144]: also add logs and call the releaseLastEvent() on super.onFailure()?

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/task/subtask/processor/PipeProcessorSubtask.java [hunk 4]
```diff
@@ -112,11 +112,13 @@ protected synchronized boolean executeOnce() throws Exception {
...
-              + "whether the implementation of PipeProcessor is correct "
-              + "according to the pipe-api description.",
-          e);
+      if (!isClosed.get()) {
+        throw new PipeException(
+            "Error occurred during executing PipeProcessor#process, perhaps need to check "
+                + "whether the implementation of PipeProcessor is correct "
+                + "according to the pipe-api description.",
+            e);
+      }
```
Comment (SteveYurongSu) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/task/subtask/processor/PipeProcessorSubtask.java:121]: else ..

Comment (DanielWang2035) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/task/subtask/processor/PipeProcessorSubtask.java:124]: changed

**Review by SteveYurongSu — APPROVED** [review]

**Pull request merged**